king/king.wdl

```diff
@@ -0,0 +1,425 @@
+version 1.0
+
+# king.wdl -- This workflow uses the KING --ibdseg method to estimate kinship coefficients,
+# and returns results for pairs of related samples. These kinship estimates can be used as
+#  measures of kinship in PC-AiR.
+
+# [1] gds2bed -- convert gds file to bed file
+
+task gds2bed {
+	input {
+		File gds_file
+
+		# optional
+		String? bed_file # provides the file name for the processed bed file, if not
+		# provided, the file name of the input gds file is used
+		File? sample_include_file
+		File? variant_include_file
+		
+		# runtime attributes
+		Int addldisk = 5
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	# Estimate disk size required
+	Int gds_size = ceil(size(gds_file, "GB"))
```

Comment:
It is unlikely for users to have a large sample_include_file or a large variant_include_file, but is technically possible. Add some calculations for those files. Since they're optional, you can use select_first(). 

Let me just save you some time here: combining select_first() with size() and ceil() is kind of a pain, and it seems this is the only way to do it:

Int smpl_size = select_first([ceil(size(sample_include_file, "GB")), 0])

ie, select_first, then ceil, then size. Try to do it in any other order and it will probably break. In this scenario, if sample_include_file is not defined, then in theory smple_size will be 0.

In addition to accounting for the sample include and the variant include, don't forget about your output file. Do you expect your bim, bam, and fam to total less than the size of your input GDS? If so, this is easy, just double the size of your GDS input in your calculation of final_disk_size. This will be trickier to estimate if you expect your outputs to be bigger than your input... try to gauge a good ratio from your test runs and based on that improve the formula for final_disk_size. It doesn't have to be perfect, especially as users can slap on a few extra gigs with addldisk, but as-written this is likely to fail on "real" data which is larger than what we're testing with.